### PR TITLE
Address a few TODOs

### PR DESCRIPTION
**Pull request by Vrixyz**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ the rust programming language.
 #![allow(clippy::type_complexity)] // Complains about closures that are fairly simple.
 #![doc(html_root_url = "http://docs.rs/parry/0.1.1")]
 #![cfg_attr(not(feature = "std"), no_std)]
-#![cfg_attr(not(feature = "rkyv"), deny(unused_qualifications))] // TODO: deny that everytime
+#![deny(unused_qualifications)] // TODO: deny that everytime
```
Comment (Vrixyz) [inline src/lib.rs:23]: I believe that's fixed by https://github.com/dimforge/parry/pull/273/files#diff-539eccbdd3d52ed1a28671920995e8af3f052e319dabb925b39308a924bc5231R126

File: src/mass_properties/mass_properties_convex_polygon.rs [hunk 2]
```diff
@@ -1,5 +1,3 @@
-#![allow(dead_code)] // TODO: remove this
```
Comment (Vrixyz) [inline src/mass_properties/mass_properties_convex_polygon.rs:1]: I'm not sure why it was necessary ; I'm thinking some feature flag or just a temporary addition while the code wasn't used :thinking:

File: src/query/epa/epa3.rs [hunk 3]
```diff
@@ -375,6 +375,8 @@ impl EPA {
                     let new_face_id = self.faces.len();
                     let new_face;
 
+                    // TODO: Thierry: We can probably remove that scope now, but I prefer to discuss it first.
+                    //                I assume NLL is for Non Lexical Lifetimes.
                     // TODO: NLL
                     {
                         let face_adj = &mut self.faces[edge.face_id];
```
Comment (Vrixyz) [inline src/query/epa/epa3.rs:382]: open question :thought_balloon:

File: src/query/ray/ray.rs [hunk 4]
```diff
@@ -79,7 +79,7 @@ pub struct RayIntersection {
     /// Otherwise, the normal points outward.
     ///
     /// If the `time_of_impact` is exactly zero, the normal might not be reliable.
-    // TODO: use a Unit<Vector> instead.
+    // TODO: use a Unit<Vector> instead. // TODO: Thierry: should we use Unit for [`Ray::dir`] too ?
     pub normal: Vector<Real>,
```
Comment (Vrixyz) [inline src/query/ray/ray.rs:83]: not entirely trivial to do ; and will have API impacts

File: src/lib.rs [hunk 5]
```diff
@@ -11,7 +11,7 @@ the rust programming language.
 #![deny(unused_parens)]
 #![deny(non_upper_case_globals)]
 #![deny(unused_results)]
-#![warn(missing_docs)] // TODO: deny this
+#![deny(missing_docs)]
```
Comment (sebcrozet) [inline src/lib.rs:14]: Let’s keep this a `warn` and remove the `TODO`. I find it more convenient to just warn locally, and have it fail only on CI by denying warnings.

File: src/mass_properties/mass_properties_convex_polygon.rs (hunk 2 above)
Comment (sebcrozet) [reply src/mass_properties/mass_properties_convex_polygon.rs:1]: Yeah, that sounds likely.

**Review by sebcrozet — COMMENTED** [review]
Looks good, thanks! This needs CHANGELOG entries to indicate the (breaking) changes.

File: src/query/intersection_test/intersection_test_composite_shape_shape.rs [hunk 6]
```diff
@@ -118,92 +114,3 @@ where
         SimdVisitStatus::MaybeContinue(mask)
     }
 }
-
-/// A visitor for checking if a composite-shape and a shape intersect.
-#[deprecated(note = "Use IntersectionCompositeShapeShapeVisitor instead.")]
-pub struct IntersectionCompositeShapeShapeBestFirstVisitor<'a, D: ?Sized, G1: ?Sized + 'a> {
```
Comment (Vrixyz) [inline src/query/intersection_test/intersection_test_composite_shape_shape.rs:124]: breaking change

**Comment by wlinna** [comment]
Could we also have a PR for rapier that addresses the removal of `IntersectionCompositeShapeShapeBestFirstVisitor`? Rapier still uses it and I can't compile  the git / local version of it anymore because of this change.

**Comment by Vrixyz** [comment]
Thanks @wlinna for bringing this to our attention!

- I opened 2 PRs to help and discuss this change:
  - https://github.com/dimforge/parry/pull/290
  - https://github.com/dimforge/rapier/pull/763

**Pull request merged**